cmd/all-in-one/main.go

```diff
@@ -255,7 +259,13 @@ func startCollector(
 
 	{
 		r := mux.NewRouter()
-		apiHandler := collectorApp.NewAPIHandler(jaegerBatchesHandler)
+		grpcBuilder.CollectorHostPorts = append(grpcBuilder.CollectorHostPorts, fmt.Sprintf("127.0.0.1:%d", cOpts.CollectorGRPCPort))
+		cp, err := agentApp.CreateCollectorProxy(repOpts, tchanBuilder, grpcBuilder, logger, metricsFactory)
```

Comment:
collector is already running the modules that generate configurations, there is no reason for the new HTTP handler to do another network hop back to collector itself, we should be able to just give the handler an implementation of an interface that provides the data directly